frame/session/src/lib.rs

```diff
@@ -589,6 +596,12 @@ impl<T: Config> Module<T> {
 		let validators = session_keys.iter()
 			.map(|(validator, _)| validator.clone())
 			.collect::<Vec<_>>();
+		// Note this should never truncate because queued keys is also bounded to `MaxValidators`,
+		// but we do so defensively.
```

Comment:
If `BoundedVec` add a `map` method we could use it here and guarantee at compile-time that we're getting a `BoundedVec` with the same size bound.